third_party/move/tools/move-asm/src/syntax.rs

```diff
@@ -0,0 +1,783 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Abstract syntax and parser of the assembler language.
+//!
+//! Move assembler files (ending in `.masm`) have the following
+//! EBNF syntax. Notice that this grammar use indentation
+//! expressed as `IND` and newline, denoted as `LF` in the syntax.
+//! Moreover, we use `ID` for a simple identifier, and `QID` for
+//! a qualified identifier, an optional address constant followed
+//! by a sequence of qualified identifiers separated by `::` (as in
+//! `0x66::bar::foo`, or `bar::foo`).
+//!
+//! ```
+//! unit :=
+//!   { address_alias LF }
+//!   ( "module" QID | "script" ) LF
+//!   { struct_def | fun_def }
```

Comment:
How do we declare `friend` modules?

Reply:
This and multiple more features are missing.